### PR TITLE
replace deprecated set-output command

### PR DESCRIPTION
**Pull request by netanelrabinowitz**
Replace all reference to deprecated `set-output` command `::set-output name= ...` in the README, the examples file
and the`Tests` workflow. #953

### CODE REVIEW

**Comment by bishal-pdMSFT** [comment]
The [docs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter) don't seem to suggest such difference. @rentziass is this difference in behaviour across Windows and Linux expected?

> Previous Test run failed because the different syntax needed for windows to read the `GITHUB_OUTPUT` so instead of introducing a new step, something like the snippet bellow, I just evaluate the npm cache directory directly in the `path` input
> 
> ```
> - name: Determine npm cache directory windows
>     if: runner.os == 'Windows'
>     run: echo "dir=$(npm config get cache)" >> $env:GITHUB_OUTPUT
> - name: Determine npm cache directory non-windows
>     if: runner.os != 'Windows'
>     run: echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
> ```

**Comment by rentziass** [comment]
Hey @bishal-pdMSFT 👋  yeah it's there but you need to switch from Bash to PowerShell at the very top of the page :)

**Comment by bishal-pdMSFT** [comment]
>you need to switch from Bash to PowerShell at the very top of the page

@rentziass at top of page?? Do you mean to use `pwsh` as shell?

**Comment by heddn** [comment]
The composer docs work correctly.

**Comment by TheNorthMemory** [comment]
Hi there, another choice via [here](https://docs.github.com/en/actions/using-jobs/setting-default-values-for-jobs)

> Use `defaults` to create a `map` of default settings that will apply to all jobs in the workflow. 

Placed the following `yml` for the workflow,

```yaml
defaults:
  run:
    shell: bash
```

An full example can be found [here](https://github.com/TheNorthMemory/xml/blob/main/.github/workflows/ci.yml)

Hopes it may helpful.

File: examples.md [hunk 1]
```diff
@@ -316,7 +316,7 @@ If using `npm config` to retrieve the cache directory, ensure you run [actions/s
 - name: Get npm cache directory
   id: npm-cache-dir
   run: |
-    echo "::set-output name=dir::$(npm config get cache)"
+    echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
```
Comment (voxpelli) [inline examples.md:319]: I sadly don't think this works on Windows

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -25,14 +25,10 @@ jobs:
...
         node-version: 16.x
-    - name: Determine npm cache directory
-      id: npm-cache
-      run: |
-        echo "::set-output name=dir::$(npm config get cache)"
     - name: Restore npm cache
       uses: actions/cache@v3
       with:
-        path: ${{ steps.npm-cache.outputs.dir }}
+        path: $(npm config get cache)
```
Comment (voxpelli) [inline .github/workflows/workflow.yml:31]: This seems to give the following error in the action:

> Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.

**Comment by Togopeva13** [comment]
> Previous Test run failed because the different syntax needed for windows to read the `GITHUB_OUTPUT` so instead of introducing a new step, something like the snippet bellow, I just evaluate the npm cache directory directly in the `path` input
> 
> ```
> - name: Determine npm cache directory windows
>     if: runner.os == 'Windows'
>     run: echo "dir=$(npm config get cache)" >> $env:GITHUB_OUTPUT
> - name: Determine npm cache directory non-windows
>     if: runner.os != 'Windows'
>     run: echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
> ```

**Pull request closed without merging**